### DISCUSSION TITLE
Attribute Error at testing endpoints

### DISCUSSION THREAD

**Discussion by agustin-del-pino** [fastapi/fastapi#6251] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from models.odm import MyBeanieModel

@router.post("/endpoint")
async def endpoint(request: Request):
    await MyBeanieModel.find_one(MyBeanieModel.SomeAttr == request.some_from_req)


# Test
client_test = TestClient(application)

@pytest.fixture(autouse=True)
async def mock_db():
    client = AsyncMongoMockClient()
    application.db = client.get_database(name="db")
    await init_beanie(document_models=[MyBeanieModel], database=application.db)


def test_endpoint():
    test_client.post("/endpoint")
```

### Description

Hello,

The issue I've was already reported [here](https://github.com/tiangolo/fastapi/issues/4551) but there no answer of how to solve it. I'm using Beanie + Motor, for DB and ODM management. 

The exception is this:
```
E       AttributeError: type object 'SomeAttr' has no attribute 'app_identifier'
```

I really don't know what part is the responsable of this exception. I mean, could be Motor (MotorMock for test), Beanie or FastAPI, but couldn't figure out 😔. I just post this here because it was reported in the past. 

In runtime works like charm, but when at testing execution blows up.

Thanks!.

### Operating System

macOS

### FastAPI Version

0.78.0

### Python Version

3.10

**Comment by ArtemZaitsev1994** [comment]
The same =( 
Any updates?

**Comment by TheRealLorenz** [comment, marked as answer]
Hi there, in my case the problem was caused by the @app.on_event("startup) decorator which didn't execute my init function. I solved it by wrapping the TestClient instance with a `with` statement, like this:

```python
with TestClient(app) as client:
    # do stuff
```

For further information: https://fastapi.tiangolo.com/advanced/testing-events/

**Comment by tiangolo** [comment]
Thanks @agustin-del-pino for coming back to close the issue! Yep, that's not in FastAPI, there's nothing internally with that name.

And thanks for the help @TheRealLorenz ! 🍰
